File: src/ssr_pkg/src/chao_node.cpp
#include <ros/ros.h>
#include<std_msgs/String.h>

int main(int argc, char  *argv[])
{
    ros::init(argc, argv, "chao_node");
    ros::NodeHandle nh;
    ros::Publisher  pub =  nh.advertise<std_msgs::String>("my_chao_topic", 10);
    ros::Rate loop_rate(10);

    while (ros::ok())
    {
        printf("chao!\n");
        std_msgs::String msg;
        msg.data = "msg_of_chao";
        pub.publish(msg);
        loop_rate.sleep();
    }

    return 0;
} 

File: src/ssr_pkg/scripts/chao_node.py
#!/usr/bin/env python
# encoding:utf8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import carry
if __name__ == "__main__":
    rospy.init_node("chao_node")
    rospy.logwarn("chao!py!")

    pub = rospy.Publisher("my_chao_topic", carry, queue_size=10)
    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        rospy.loginfo("chao!!!py")
        msg = carry()
        msg.data = "msg_of_chao_py"
        msg.star = 10
        msg.grade = "wangzhe"
        pub.publish(msg)
        rate.sleep()